src/app/admin/products/[productId]/edit/page.tsx

```diff
@@ -0,0 +1,779 @@
+"use client";
+import { useParams } from "next/navigation";
+import { doc, getDoc } from "firebase/firestore";
+import { db } from "../../../../../utils/firebase";
+import { useEffect, useState, useActionState } from "react";
+import { Button } from "../../../../components/Button";
+import {
+  allCategories,
+  allTags,
+  allAvailabilityStatus,
+  allReturnPolicies,
+  Product,
+} from "../../../../../types/types";
+import { editProduct } from "../../../../actions/admin/products";
+import Form from "next/form";
+import { useForm } from "react-hook-form";
+
+const initialState: EditProductFormState = {
+  success: false,
+  inputs: {},
+  errors: {},
+};
+
+export interface EditProductFormState {
+  success: boolean;
+  message?: string;
+  inputs?: Partial<Product>;
+  errors?: {
+    [K in keyof Product]?: string[];
+  };
+  data?: any;
+}
+
+export default function EditProduct() {
+  const {
+    register,
+    formState: { errors },
+  } = useForm({ mode: "all" });
+  const [state, formAction, isPending] = useActionState<
+    EditProductFormState,
+    FormData
+  >(editProduct, initialState);
+
+  const params = useParams<{ productId: string }>();
+  const [product, setProduct] = useState<any>();
+
+  useEffect(() => {
+    async function fetchData() {
+      const docRef = doc(db, "products", params.productId);
+      const docSnap = await getDoc(docRef);
+      if (docSnap.exists()) {
+        setProduct(docSnap.data());
+      } else {
+        console.log("No such document!");
+      }
+    }
+    fetchData();
+  }, []);
+
+  if (isPending) return <p>Loading...</p>;
+  return (
+    <main className="max-w-4xl mx-auto my-6">
+      {product ? (
+        <div>
+          <Form
+            action={formAction}
+            className="grid grid-cols-2 gap-6 bg-white p-8 rounded-xl shadow-md"
+          >
+            <div className="flex flex-col col-span-2">
+              <label htmlFor="productId" className="font-bold mb-1">
```

Comment:
ID should not be changeable, remove it from the form completely. You can display it as plain text, but no need to use input here.